### PR TITLE
WIP: Sponsor this ebook button on the books page 

### PR DESCRIPTION
**Pull request by tabshaikh**
**Description**: What does this PR achieve? [feature]
Shows the `Sponsor this ebook` to a specific group of sponsors and then after the sponsor clicks 
the button opens up a modal which then redirects the user to https://ia801409.us.archive.org/20/items/book-sponsorship-dev/index.html 

Closes # (Specific Issue to be made) #659 (complete issue)

Points to be noted: 
Just a rough implementation, not ready for review but just for checking functionality.
Passing the parameters from the books page to book-sponsorship page left.
Black button implemented for now, would change to the decided button later

### CODE REVIEW

**Comment by tabshaikh** [comment]
@mek finished the second point and a bit more 😉 just want you to pull locally the 659/feature/book-sponsorship branch and check the flow out

File: openlibrary/macros/LoanStatus.html [hunk 1]
```diff
@@ -80,9 +80,60 @@
     <a href="$viewbook.replace('XXX', page.ocaid)" title="Use BookReader to read online" class="cta-btn cta-btn--available">Read</a>
 
 $elif (not page.get('ocaid') or page.is_access_restricted()) and editions_page:
-    <form>
-      <input type="submit" class="cta-btn" disabled value="No ebook available">
-    </form>
+    <!-- If user belongs to sponser-book group then show the sponser available button  -->
```
Comment (mekarpeles) [inline openlibrary/macros/LoanStatus.html]: typo (sponsor)

File: openlibrary/macros/LoanStatus.html [hunk 2]
```diff
@@ -80,9 +80,60 @@
...
+    <!-- To implement: Using the policy api add more constraint that the book which is sponsorable
+    should only show this type of button  -->
+    $if (user and user.is_sponsor()):
+      <button id="myBtn" class="cta-btn" style="background-color: black">Sponser this ebook</button>
```
Comment (mekarpeles) [inline openlibrary/macros/LoanStatus.html]: Sponser -> Sponsor
Let's also capitalize the B (to give the text left-right balance) 
`Sponsor this eBook`

File: static/css/components/modal.less [hunk 3]
```diff
@@ -0,0 +1,38 @@
+/* The Modal (background) */
+.modal {
```
Comment (mekarpeles) [inline static/css/components/modal.less]: Don't we have an existing modal we can use?
e.g. the one for adding book-covers or creating a new list?

File: openlibrary/macros/LoanStatus.html (hunk 2 above)
Comment (tabshaikh) [reply openlibrary/macros/LoanStatus.html]: Yup silly mistake

File: static/css/components/modal.less (hunk 3 above)
Comment (mekarpeles) [reply static/css/components/modal.less]: Moving our modal to its own component file seems like a reasonable idea, but I don't think we want to create a new modal.

@jdlrobson thoughts/tips on how you'd like us to approach this?

Comment (tabshaikh) [reply static/css/components/modal.less]: Don't know about that 🤔

File: openlibrary/macros/LoanStatus.html (hunk 1 above)
Comment (tabshaikh) [reply openlibrary/macros/LoanStatus.html]: Again here too sorry :)

**Comment by tabshaikh** [comment]
@mek this is not ready for a review there are inline styles too just wanted you to test it 😀 if it works completely according to our expectation then I would refine the pr

File: static/css/components/modal.less (hunk 3 above)
Comment (tabshaikh) [reply static/css/components/modal.less]: The modal we have is somewhat broken 
Screenshot from 2019-03-19 12-34-47
I have implemented something like this 
Screenshot from 2019-03-16 02-11-43

File: openlibrary/macros/LoanStatus.html (hunk 1 above)
Comment (tabshaikh) [reply openlibrary/macros/LoanStatus.html]: Done

File: static/css/components/modal.less (hunk 3 above)
Comment (jdlrobson) [reply static/css/components/modal.less]: $.colorbox is the modal we use. I think we should continue to use it for the time being to build lightboxes (however we might want to consider redesigning them and improving the CSS! :-)

The close icon spilling out of the modal is fixed in https://github.com/internetarchive/openlibrary/pull/1997

See http://www.jacklmoore.com/colorbox/

File: openlibrary/macros/LoanStatus.html [hunk 4]
```diff
@@ -80,9 +80,60 @@
...
+      <button id="myBtn" class="cta-btn" style="background-color: black">Sponsor this eBook</button>
+      <!-- The Modal -->
+      <div id="myModal" class="modal">
+          
+        <!-- Modal content -->
+        <div class="modal-content" style="text-align:center">
+          <span class="close">&times;</span>
+          <p>
+            <h1>
+                Unlock this Book
```
Comment (mekarpeles) [inline openlibrary/macros/LoanStatus.html]: Let's make this "Sponsor this Book"

Comment (tabshaikh) [reply openlibrary/macros/LoanStatus.html]: Okay doing it

**Comment by tabshaikh** [comment]
@mekarpeles please can you update the pr with the latest changes

**Pull request closed without merging**